src/main/java/com/xceptance/neodymium/util/SelenideAddons.java

```diff
@@ -410,8 +411,21 @@ public static void wrapAssertionError(final Runnable runnable)
      */
     public static void dragAndDrop(SelenideElement elementToMove, int horizontalMovement, int verticalMovement)
     {
-        // perform drag and drop via the standard Selenium way
-        new Actions(Neodymium.getDriver()).dragAndDropBy(elementToMove.getWrappedElement(), horizontalMovement, verticalMovement).build().perform();
+        try
+        {
+            // perform drag and drop via the standard Selenium way
+            new Actions(Neodymium.getDriver()).dragAndDropBy(elementToMove.getWrappedElement(), horizontalMovement, verticalMovement).build().perform();
+        }
+        catch (MoveTargetOutOfBoundsException targetOutOfBound)
+        {
+            String parameterMessage = horizontalMovement != 0 ? (verticalMovement != 0 ? "'horizontalMovement' and 'verticalMovement'" : "'horizontalMovement'")
+                                                              : "'verticalMovement'";
+
+            throw UIAssertionError.wrap(WebDriverRunner.driver(),
+                                        new AssertionError("Target out of bounds. Try to decrease the absolute value of " + parameterMessage
```

Comment:
Please adjust the error message: 
"Performing drag and drop with an element moved the element out of the viewport. Try to scroll the element completely into the view port or to decrease the absolute values of your movements."
A special treatment for each direction within the error message is not required.

Reply:
Done